packages/core/src/cli/commands/build.command.ts

```diff
@@ -0,0 +1,135 @@
+
+  const workspace = ctx.workspace!;
+  const service = ctx.selectedService;
+  checkForSchemaErrors(workspace);
+
+  if (!ctx.configLoader.workspaceInfo.isMonorepo) {
+    throw new CliExitError('`dmno build` only works in monorepos');
+  }
+
+  const buildDirPath = `${service.path}/.dmno-built`;
+  // delete existing build folder
+  try {
+    await rm(buildDirPath, { recursive: true });
+  } catch (err) {}
+
+  if (!ctx.workspace) throw new Error('workspace not loaded');
+
+  // TODO: do we want to write a top level json file with metadata?
+  const buildMetadata: DmnoBuildInfo = {
+    isMonorepo: ctx.configLoader.workspaceInfo.isMonorepo,
+    rootService: ctx.workspace.rootService.serviceName,
+    selectedService: ctx.selectedService.serviceName,
+  };
+
+  const buildMeta = ctx.workspace.getServiceMetaForBuild(service.serviceName);
+
+
+
+  for (const workspaceService of ctx.workspace?.allServices || []) {
+    // skip any unrelated services according to the DAG (related via parent or pick)
+    if (
+      workspaceService !== ctx.selectedService
+      && !buildMeta.requiredServices.includes(workspaceService.serviceName)
+    ) continue;
+
+    const configFilePath = `${workspaceService.path}/.dmno/config.mts`;
+    const buildPackageDirPath = `${buildDirPath}/${workspaceService.serviceName.replaceAll('/', '__')}`;
+    console.log('Bundling dmno config file for service ', workspaceService.serviceName);
+    console.log('> config file:', configFilePath);
+    console.log('> bundled config package dir:', buildPackageDirPath);
+
+    let bundle: RollupBuild | undefined;
+    bundle = await rollup({
+      input: configFilePath,
+      external: ['dmno', /@dmno\/.*/],
+      plugins: [
+        commonjsRollupPlugin(),
+        nodeResolve({
+          preferBuiltins: true, // not sure about this - but false and undefined show warnings
+        }),
+
+        // typescriptRollupPlugin({
+        //   include: ['**/*.{mts,ts}'],
+        // }),
+
+        // alternative to @rollup/plugin-typescript which does faster builds without typechecking
+        sucraseRollupPlugin({
+          include: ['**/*.{mts,ts}'],
+          exclude: ['node_modules/**'],
+          transforms: ['typescript'],
+        }),
+      ],
+    });
+
+    // write built JS config file
+    await bundle.write({
+      file: `${buildPackageDirPath}/.dmno/config.js`,
+    });
+    await bundle?.close();
+
+
+    // copy vault file(s)
+    // TODO: will need to figure out how to generalize this
+    // we could copy everything that is not git-ignored?
+    // we could let plugins specify a list of patterns to copy
+    await exec(`cp ${workspaceService.path}/.dmno/*.vault.json ${buildPackageDirPath}/.dmno`);
+
+    // copy package.json file
+    await copyFile(`${workspaceService.path}/package.json`, `${buildPackageDirPath}/package.json`);
+  }
+
+  await writeFile(`${buildDirPath}/dmno-build-info.json`, JSON.stringify(buildMetadata, null, 2), 'utf8');
```

Comment:
Not much in this file yet (see above) but it at least gives us somewhere to put extra data.